### DISCUSSION TITLE
Customize Quarkus content negotiation

### DISCUSSION THREAD

**Discussion by Treiblesschorle** [quarkusio/quarkus#44370] [Q&A] [answered]
Hi, I'm trying to implement rfc9457 style error handling for a service, basically returning error response bodies in json format using the content type "application/problem+json". This should be controlled via the Accept header, i.e. if the client accepts "application/problem+json" I return the error json and a plain text error otherwise (this is also important for backwards compatibility), which I implemented in an `ExceptionMapper`. The service has many endpoints and I think I would need to add "application/problem+json" to the `@Produces` of all of them, otherwise Quarkus sends a 406. Is it instead somehow possible to configure the content negotiation somehow globally to make all endpoints accept/produce this media type?

**Comment by gsmet** [comment, marked as answer]
I'm not sure I fully understand your problem because I would expect your clients to say that they support both `application/json` and `application/problem+json` with the latter having a lower priority.

With this in mind, I wouldn't expect any issue when negotiating the content type? Or does Quarkus checks the `Content-Type` also when pushing the Response? I would have expected that you could do whatever you want in the response.

If what I'm saying doesn't make any sense, maybe a small reproducer could help so that we could have a look at what's going on?

**Reply by Treiblesschorle** [reply]
> I'm not sure I fully understand your problem because I would expect your clients to say that they support both application/json and application/problem+json with the latter having a lower priority.

That makes sense indeed. Missed that somehow.
